py-polars/polars/_utils/various.py

```diff
@@ -56,7 +61,7 @@ def _process_null_values(
         return null_values
 
 
-def _is_generator(val: object) -> bool:
+def _is_generator(val: object | Iterator[T]) -> TypeIs[Iterator[T]]:
```

Comment:
> It would be pretty easy - if you just change the return type of `_is_generator ` to `TypeGuard[Generator[Any, Any, Any]]`, you should be able to remove the ignore. 

@max-muoto I was thinking more to use `TypeIs`, since that can preserve the inner type.

Reply:
Ah yes, `TypeIs` is generally a better choice.